internal/webhook/external/azure.go

```diff
@@ -0,0 +1,219 @@
+package external
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	"github.com/akuity/kargo/internal/git"
+	"github.com/akuity/kargo/internal/helm"
+	xhttp "github.com/akuity/kargo/internal/http"
+	"github.com/akuity/kargo/internal/image"
+	"github.com/akuity/kargo/internal/logging"
+)
+
+const (
+	azureSecretDataKey = "secret"
+	azure              = "azure"
+)
+
+func init() {
+	registry.register(
+		azure,
+		webhookReceiverRegistration{
+			predicate: func(cfg kargoapi.WebhookReceiverConfig) bool {
+				return cfg.Azure != nil
+			},
+			factory: newAzureWebhookReceiver,
+		},
+	)
+}
+
+// azureWebhookReceiver is an implementation of WebhookReceiver that handles
+// inbound webhooks from both Azure Container Registry (ACR) and Azure DevOps.
+type azureWebhookReceiver struct {
+	*baseWebhookReceiver
+}
+
+// newAzureWebhookReceiver returns a new instance of azureWebhookReceiver.
+func newAzureWebhookReceiver(
+	c client.Client,
+	project string,
+	cfg kargoapi.WebhookReceiverConfig,
+) WebhookReceiver {
+	return &azureWebhookReceiver{
+		baseWebhookReceiver: &baseWebhookReceiver{
+			client:     c,
+			project:    project,
+			secretName: cfg.Azure.SecretRef.Name,
+		},
+	}
+}
+
+// getReceiverType implements WebhookReceiver.
+func (a *azureWebhookReceiver) getReceiverType() string {
+	return azure
+}
+
+// getSecretValues implements WebhookReceiver.
+func (a *azureWebhookReceiver) getSecretValues(
+	secretData map[string][]byte,
+) ([]string, error) {
+	secretValue, ok := secretData[azureSecretDataKey]
+	if !ok {
+		return nil,
+			errors.New("secret data is not valid for an Azure WebhookReceiver")
+	}
+	return []string{string(secretValue)}, nil
+}
+
+// getHandler implements WebhookReceiver.
+func (a *azureWebhookReceiver) getHandler(requestBody []byte) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		logger := logging.LoggerFromContext(ctx)
+		logger.Debug("received azure event", "userAgent", r.UserAgent())
+
+		switch {
+		// format is AzureContainerRegistry/<version>
+		case strings.Contains(r.UserAgent(), "AzureContainerRegistry"):
+			a.handleAcrEvent(ctx, w, requestBody)
+		// Format is VSServices/<version>
+		case strings.Contains(r.UserAgent(), "VSServices"):
+			a.handleAzureDevOpsEvent(ctx, w, requestBody)
+		default:
+			xhttp.WriteErrorJSON(
+				w,
+				xhttp.Error(
+					errors.New(
+						"request does not appear to have originated from a supported service",
+					),
+					http.StatusBadRequest,
+				),
+			)
+			return
+		}
+	})
+}
+
+func (a *azureWebhookReceiver) handleAcrEvent(
```

Comment:
The widely observed convention in Go is to not mix case on an acronyms. If it starts small, keep going with small. If it starts, big, keep going with big.


```suggestion
func (a *azureWebhookReceiver) handleACREvent(
```